### PR TITLE
Passkey example application: A Major refactoring

### PR DESCRIPTION
**Pull request by mmv08**
This PR lays the foundation for https://github.com/safe-global/safe-modules/issues/360 and https://github.com/safe-global/safe-modules/issues/361.

**Motivation**
The example application was built as a mere example, and all of the state/screen management was done in a single file—` src/App.ts`—through tens of different boolean and state variables. Since we have now decided to add more features to the app, the approach is no longer scalable.

**Solution**
Isolate every step in the deployment flow - in other words, make every step in a process its route. This should also help the developers to comprehend the example since now, to understand what is needed to create a passkey, they only have to review the `CreatePasskey` route, not mixed with anything else.

**This PR**
- Add `react-router-dom` to manage the routing in the application
- Create a route for every step in the process
- Introduce a new Safe Dashboard page with a link to the Safe{Wallet} App (This page will be used for #360, the new action will be added here)

**Demo**
https://www.loom.com/share/b5dcec589cc44fd8be8cefe1ef9cb306

### CODE REVIEW

File: examples/4337-passkeys/src/routes/CreatePasskey.tsx
Comment (nlordell) [inline examples/4337-passkeys/src/routes/CreatePasskey.tsx:42]: Maybe add a blurb about how the credential will be used (i.e. it will be used to verify user operations on chain, blah, blah, etc,) - in other words give more context as to _why_ we are creating the passkey.

File: examples/4337-passkeys/src/main.tsx
Comment (nlordell) [inline examples/4337-passkeys/src/main.tsx:18]: Shouldn't this be:
```suggestion
    path: HOME,
```

Comment (nlordell) [reply examples/4337-passkeys/src/main.tsx:18]: (disclaimer: I'm react noob)

File: examples/4337-passkeys/src/routes/Home.tsx
Comment (nlordell) [inline examples/4337-passkeys/src/routes/Home.tsx:61]: Why do you sometimes redirect in the loader, but for other cases redirect in the component?

Comment (nlordell) [reply examples/4337-passkeys/src/routes/Home.tsx:61]: Is it because this comes from the `useCodeAtAddress(walletProvider, safeAddress)` which is `async`?

**Review by nlordell — APPROVED** [review]
LGTM. Just one question.

File: examples/4337-passkeys/src/routes/Home.tsx
Comment (mmv08) [reply examples/4337-passkeys/src/routes/Home.tsx:61]: That's correct, but probably not for the reason you think. For the async data, we need to use the user's RPC provider because we don't have our own one, but the `web3modal` doesn't expose a reliable method to get it except a react hook. I tried to use the `web3modal` object directly, but it was not reliable and would fail in cases where the user doesn't have a wallet connected.

There might be a workaround where we retrieve the provider higher up in the component hierarchy and pass it to the router itself. However, I decided not to waste time on trying that since it's still primarily an example app :smile:

File: examples/4337-passkeys/src/routes/CreatePasskey.tsx
Comment (mmv08) [reply examples/4337-passkeys/src/routes/CreatePasskey.tsx]: Check the new copy :)

**Comment by mmv08** [comment]
Just so you know, I requested a review from a React expert, @iamacook. He said he would have time to check tomorrow, so let's hold off on the merge until he approves it.

File: examples/4337-passkeys/src/routes/DeploySafe.tsx
Comment (mmv08) [reply examples/4337-passkeys/src/routes/DeploySafe.tsx]: Unfortunately, this doesn't work because the loader/useLoaderData doesn't accept any type arguments, and the return type of the loader function is either the data or a response type (redirect, for example)

File: examples/4337-passkeys/src/routes/constants.ts
Comment (mmv08) [reply examples/4337-passkeys/src/routes/constants.ts]: done (although i think it was enough cos it was reported from the routes "package")

Comment (nlordell) [reply examples/4337-passkeys/src/routes/constants.ts]: > although i think it was enough cos it was reported from the routes "package")

I tend to agree.

You could express it as:

```ts
import * as routes from "path/to/routes.ts"

routes.HOME
routes.SAFE
// etc.
```

**Pull request merged**
